### ISSUE TITLE
Support for getting page session source

### ISSUE THREAD

**Issue by steve0hh** [keathley/wallaby#79]
Was wondering if you guys are ok with adding an addition API to get the page source.

This would aid developers that would like to stick with their preferred DSL like [floki](https://github.com/philss/floki).

I already have the function working in my fork of this project but would need some guidance in writing the tests. (where to place them etc..)

Example of the proposed API:

```
iex(4)> session |> Wallaby.Driver.visit("http://www.example.com") |> Wallaby.Driver.source()

"<!DOCTYPE html><html><head>\n    <title>Example Domain</title>\n\n 
the page source of the session currently
</html>"
```

**Comment by keathley** [comment]
Thanks @steve0hh. A PR for this would be great. If you wanna submit what you have I can take a look and we can talk through any missing details.

I'm also curious to know a little more about how you're using floki. Is it specific to your use case? Are there specific pieces of functionality that are missing from Wallaby? Or is it just easier to manage then Wallaby's DSL? Looks like a cool project!

**Comment by steve0hh** [comment]
No problem. I am pretty new to the elixir.

I need to build a automation tool that does some scraping.

Was thinking if this was a good use case. 😄  however, there's some issue with phantomjs not being able to render some of the sites I need to scrape. I might need to use selenium.

**Comment by keathley** [comment]
Are they sites that use a lot of JS? They may be using new JS that phantom doesn't understand (and throwing errors) or they may just be taking a long time to load. I'm not sure if floki is doing any blocking but my suspicion is that it isn't. You can try the same query with the Wallaby DSL first which will block until the element appears on the page and see if that fixes your issue.

**Comment by steve0hh** [comment]
Yes. It's basically an entire page driven by emberjs. (My FYP project)

> You can try the same query with the Wallaby DSL first which will block until the element appears on the page and see if that fixes your issue

I tried it too. It seems to be the problem with `phantomjs` not wallaby. 

Things I did:
- Suppressed the javascript errors (Complaining about a module not being able to load)
- Set timeout to infinite
- Used the `find()` (Couldn't find element)
- Took a screenshot and found out that the javascript wasn't running correctly

Another reason why I wanted to have the `source` function, is also because I wanted to scrape the HTML in the nodes too. 

Spent an afternoon looking through your code and editing it and thought that you guys might be ok with a PR hence this thread. 😄
